yarn-project/prover-node/src/prover-node-publisher.test.ts

```diff
@@ -118,16 +124,64 @@ describe('prover-node-publisher', () => {
       toBlock: 40,
       expectedPublish: true,
     },
+
+    // We create a case that will need to lower the anchor forcefully to ensure it will do so
+    {
+      pendingBlockNumber: 65n,
+      provenBlockNumber: 32n,
+      fromBlock: 33,
+      toBlock: 64,
+      expectedPublish: true,
+      message: '',
+      mustLowerAnchor: true,
+    },
   ];
 
   test.each(testCases)(
     'submits proof for epoch with pendingBlock: $pendingBlockNumber, provenBlock: $provenBlockNumber, fromBlock: $fromBlock, toBlock: $toBlock',
-    async ({ pendingBlockNumber, provenBlockNumber, fromBlock, toBlock, expectedPublish, message }) => {
+    async ({
+      pendingBlockNumber,
+      provenBlockNumber,
+      fromBlock,
+      toBlock,
+      expectedPublish,
+      message,
+      mustLowerAnchor,
+    }) => {
       // Create public inputs for every block
       const blocks = Array.from({ length: 100 }, () => {
         return RootRollupPublicInputs.random();
       });
 
+      inbox.getLastAnchorBlockNumber.mockResolvedValue(mustLowerAnchor ? BigInt(toBlock - 1) : pendingBlockNumber);
+
+      if (mustLowerAnchor) {
+        inbox.lowerAnchorForcefully.mockResolvedValue({
+          receipt: {
+            transactionHash: '0x',
+            status: 'success',
+            blockHash: '0x',
+            blockNumber: 0n,
+            contractAddress: '0x',
+            cumulativeGasUsed: 0n,
+            gasUsed: 0n,
+            logs: [],
+            logsBloom: '0x',
+            transactionIndex: 0,
+            from: '0x',
+            to: '0x',
+            type: '0x',
+            effectiveGasPrice: 0n,
+          },
+          gasPrice: {
+            maxFeePerGas: 0n,
+            maxPriorityFeePerGas: 0n,
+          },
+        });
```

Comment:
Do we even use this receipt? Maybe it'd be easier if we just changed the return value to void?

Reply:
Was mainly to follow similar structure to what we have been doing on the `rollup.ts`.